### DISCUSSION TITLE
Why different packages for different architectures?

### DISCUSSION THREAD

**Discussion by marco-eckstein** [microsoft/winget-pkgs#132143] [Q&A] [answered]
Why would one want different packages for different architectures, e.g. `Adobe.Acrobat.Reader.32-bit` and `Adobe.Acrobat.Reader.64-bit`? Seems to me like the package creators misunderstood WinGet, but maybe I am the one who misunderstands.

**Comment by Trenly** [comment, marked as answer]
> Why would one want different packages for different architectures, e.g. `Adobe.Acrobat.Reader.32-bit` and `Adobe.Acrobat.Reader.64-bit`? Seems to me like the package creators misunderstood WinGet, but maybe I am the one who misunderstands.

Most applications have separate installers for 32-bit systems and 64-bit systems, but use a single uninstall entry so that using one installer will overwrite the existing install from the other installer . However, some applications (like Acrobat Reader, Visual C++ Redistributables, etc.) actually install different applications that can exist side by side. In order for WinGet to properly manage the applications that install side-by-side, they need different Package Identifiers. Otherwise, both would be considered the same package and `WinGet upgrade` might use the x64 installer on an x86 install, which would result in the x86 not being upgraded and a new x64 package being installed

**Reply by marco-eckstein** [reply]
I see. In case of Acrobat Reader I didn't think of the possibility that someone could want a side-by-side installation. Thanks.
